onnxruntime/core/framework/session_state_utils.cc

```diff
@@ -61,20 +61,26 @@
   OrtCallback ext_delete_cb;
   Tensor* p_tensor;
   void operator()(void*) noexcept {
-    ext_delete_cb.f(ext_delete_cb.param);
+    if (ext_delete_cb.f) {
+      ext_delete_cb.f(ext_delete_cb.param);
+    }
+
     delete p_tensor;
   }
 };
 
-// given a tensor proto with externdal data return an OrtValue with a tensor for
+// given a tensor proto with external data return an OrtValue with a tensor for
 // that data; the pointers for the tensor data and the tensor itself are owned
 // by the OrtValue's deleter
 static inline common::Status ExtDataTensorProtoToTensor(const Env& env,
                                                         const std::basic_string<PATH_CHAR_TYPE>& proto_path,
                                                         const ONNX_NAMESPACE::TensorProto& tensor_proto,
                                                         Tensor& tensor, OrtCallback& ext_data_deleter) {
   ORT_ENFORCE(utils::HasExternalData(tensor_proto));
-  ORT_ENFORCE(!proto_path.empty());
+  // TODO: Do we need a check here? It's internal only so I don't think having an empty proto_path when loading
```

Comment:
>Do we need a check here? It's internal only  [](http://example.com/codeflow?start=11&length=44)

I think we can remove it, and make sure that the error message on failure includes that empty path.

Reply:
On Windows we report just the filename. On Linux the failure would occur at a point there's an invalid file descriptor, so no filename or path.

Added something to GetExtDataFromTensorProto to ensure the full path is reported if there's an error.